CMakeLists.txt

```diff
@@ -56,31 +57,37 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 ################################################################################
 
 if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+
   # Precision-based Fortran compiler flags
   set(r8_flags "-fdefault-real-8 -fdefault-double-8") # Fortran flags for 64BIT precision
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fconvert=big-endian -ffree-line-length-none -fno-range-check -fbacktrace")
-  set(CMAKE_Fortran_FLAGS_RELEASE "-O3 -funroll-all-loops -finline-functions")
-  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -g -fcheck=bounds -ffpe-trap=invalid,zero,overflow,underflow" )
-  set(CMAKE_Fortran_LINK_FLAGS "" )
+
+  # Copied from MOM5/bin/mkmf.template.nci.gfortran
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fdefault-real-8 -ffree-line-length-none -fno-range-check -Waliasing -Wampersand -Warray-bounds -Wcharacter-truncation -Wconversion -Wline-truncation -Wintrinsics-std -Wsurprising -Wno-tabs -Wunderflow -Wunused-parameter -Wintrinsic-shadow -Wno-align-commons")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -g -W -fbounds-check")
 
 elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
+
   # Precision-based Fortran compiler flags
   set(r4_flags "-real-size 32") # Fortran flags for 32BIT precision
   set(r8_flags "-real-size 64") # Fortran flags for 64BIT precision
   set(r8_flags "${r8_flags} -no-prec-div -no-prec-sqrt")
-  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fpp -fno-alias -auto -safe-cray-ptr -ftz -assume byterecl -align array64byte -nowarn -sox -traceback")
-  set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -debug minimal -fp-model source -nowarn -qoverride-limits -qno-opt-dynamic-align -qopt-prefetch=3")
-  set(CMAKE_Fortran_FLAGS_DEBUG "-g -O0 -check -check noarg_temp_created -check nopointer -warn -warn noerrors -fpe0 -ftrapuv")
-  set(CMAKE_Fortran_LINK_FLAGS "")
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -align all")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-g3 -O2 -xCORE-AVX2 -debug all -check none")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-g3 -O0 -debug all -check -check noarg_temp_created -check nopointer -warn -warn noerrors -ftrapuv")
```

Comment:
@harshula I know you want to clean-up the flags later, but I thought it important to flag that there are some conflicts here.

`CMAKE_Fortran_FLAGS_DEBUG` is appended to `CMAKE_Fortran_FLAGS`, but they set conflicting flags (e.g.,`-warn`/`-nowarn`. Similar thing happens with `CMAKE_Fortran_FLAGS_RELEASE`.

No objections in fixing this in a different PR, but it needs to be made ASAP in my opinion.

Reply:
Please document compiler flag issues at https://github.com/ACCESS-NRI/FMS/issues/1

Reply:
Done.